src/adapters/canterbury.js

```diff
@@ -0,0 +1,255 @@
+    }
+    // Wrap everything in a try/catch in case something goes wrong
+    try {
+      // Format the data
+      let data = formatData(results);
+      if (data === undefined) {
+        return cb({message: 'Failure to parse data.'});
+      }
+      cb(null, data);
+    } catch (e) {
+      console.log(e);
+      return cb({message: 'Unknown adapter error.'});
+    }
+  });
+};
+
+/**
+ * Given fetched data, turn it into a format our system can use.
+ * @param {array} results Fetched source data and other metadata
+ * @return {object} Parsed and standarized data our system can use
+ */
+
+async function formatData (results) {
+  // filter out undefined values
+  results = results.filter(r => r[0] !== undefined);
+  let measurements = [];
+  // Legal types, for some reason the JSON converts signs and spaces into unicode,
+  // and data has wind and temperature data also, which needs to be filtered out
+  let legalParams = ['PM10', 'PM2.5', 'SO2', 'CO', 'NO2', 'O3'];
+    results.forEach(r => {
+    // console.log(r)
+    const template = {
+      city: r[1].city,
+      location: r[1].location,
+      coordinates: r[1].coordinates,
+      attribution: [{name: 'Environment Canterbury', url: 'https://ecan.govt.nz/'}],
+      averagingPeriod: {unit: 'hours', value: 0.166666} // this should be aproximate to 10 minutes, not sure if this is OK
+    };
+    // Runs through all data items for the site
+      r.forEach(d => {
+      // Gets the datemoment and correct timezone of time from item
+      const dateMoment = moment.tz(d.DateTime, 'Pacific/Auckland');
+      let measurement = Object.assign({
+        'date': {
+          utc: dateMoment.toDate(),
+          local: dateMoment.format()
+        }
+      }, template);
+      // Filters out all unwanted data, and then runs through all keys
+      Object.keys(d).filter(m => {
+        for (let p of legalParams) {
+          if (m.search(p) !== -1) {
+            return true;
+          }
+        }
+        return false;
+      }).forEach(m => {
+        // Part of key is the measurement, substring until the first letter of _, to find parameter
+        measurement.parameter = m.substr(0, m.indexOf('_'));
+        measurement.value = Number(d[m]);
+        // All data is in ug/m3, except CO, which is mg/m3
+        measurement.unit = (measurement.parameter !== 'CO') ? 'ug/m3' : 'mg/m3';
+        // Unifies measurement units and parameters before adding them
+        measurement = unifyMeasurementUnits(measurement);
+        measurement = unifyParameters(measurement);
+        measurements.push(measurement);
+      });
+    });
+  });
+  //  filter out duplicates
+  measurements = measurements.filter((m, i, a) => a.findIndex(t => (t.date.utc === m.date.utc && t.parameter === m.parameter)) === i);
+  return {
+    name: 'unused',
+    measurements: measurements
+  };
+};
+
+/* There are a lot of more stations, but they dont seem to be reporting for some reason,
```

Comment:
@majesticio is this your comment or from the original PR? is it still relevant

Reply:
still relevant